src/it/scala/io/iohk/ethereum/transactionTest/util/FixtureProvider.scala

```diff
@@ -0,0 +1,140 @@
+import io.iohk.ethereum.network.p2p.messages.PV63._
+import io.iohk.ethereum.rlp.RLPImplicitConversions._
+import io.iohk.ethereum.rlp._
+import org.spongycastle.util.encoders.Hex
+
+import scala.io.Source
+import scala.util.Try
+
+object FixtureProvider {
+
+  case class Fixture(
+    blockByNumber: Map[BigInt, Block],
+    blockByHash: Map[ByteString, Block],
+    blockHeaders: Map[ByteString, BlockHeader],
+    blockBodies: Map[ByteString, BlockBody],
+    receipts: Map[ByteString, Seq[Receipt]],
+    stateMpt: Map[ByteString, MptNode],
+    contractMpts: Map[ByteString, MptNode],
+    evmCode: Map[ByteString, ByteString]
+  )
+
+  // scalastyle:off
+  def prepareStorages(blockNumber: BigInt, fixtures: Fixture): (BlockchainStorages, NodeStorage) = {
+
+    val storages: BlockchainStorages = new BlockchainStorages {
+      override val receiptStorage: ReceiptStorage = new ReceiptStorage(EphemDataSource())
+      override val evmCodeStorage: EvmCodeStorage = new EvmCodeStorage(EphemDataSource())
+      override val blockHeadersStorage: BlockHeadersStorage = new BlockHeadersStorage(EphemDataSource())
+      override val blockNumberMappingStorage: BlockNumberMappingStorage = new BlockNumberMappingStorage(EphemDataSource())
+      override val mptNodeStorage: MptNodeStorage = new MptNodeStorage(EphemDataSource())
+      override val blockBodiesStorage: BlockBodiesStorage = new BlockBodiesStorage(EphemDataSource())
+      override val totalDifficultyStorage: TotalDifficultyStorage = new TotalDifficultyStorage(EphemDataSource())
+    }
+
+    val stateStorage: NodeStorage = new NodeStorage(EphemDataSource())
+
+    val blocksToInclude = fixtures.blockByNumber.toSeq.sortBy { case (number, _) => number }.takeWhile { case (number, _) => number <= blockNumber }
+
+    blocksToInclude.foreach { case (_, block) =>
+      storages.blockBodiesStorage.put(block.header.hash, fixtures.blockBodies(block.header.hash))
+      storages.blockHeadersStorage.put(block.header.hash, fixtures.blockHeaders(block.header.hash))
+      storages.blockNumberMappingStorage.put(block.header.number, block.header.hash)
+      fixtures.receipts.get(block.header.hash).foreach(r => storages.receiptStorage.put(block.header.hash, r))
+
+      //storages.totalDifficultyStorage
+
+      def traverse(nodeHash: ByteString): Unit = fixtures.stateMpt.get(nodeHash).orElse(fixtures.contractMpts.get(nodeHash)) match {
+        case Some(m: MptBranch) =>
+          storages.mptNodeStorage.put(m)
+          stateStorage.put(m.hash, encode(m: MptNode)(MptNode.rlpEncDec))
+          m.children.collect { case Left(MptHash(hash)) if hash.nonEmpty => hash }.foreach(traverse)
+
+        case Some(m: MptExtension) =>
+          storages.mptNodeStorage.put(m)
+          stateStorage.put(m.hash, encode(m: MptNode)(MptNode.rlpEncDec))
+          m.child match {
+            case Left(MptHash(hash)) if hash.nonEmpty => traverse(hash)
```

Comment:
You are right :+1: But shouldn't we consider that case here so that this pattern matching considers all the cases and then the matching is exhaustive?